FreeRTOS-Plus/Demo/coreSNTP_Windows_Simulator/DemoTasks/SNTPExample.c

```diff
@@ -0,0 +1,772 @@
+         * since last time synchronization. */
+        calculateCurrentTime( &systemClock.baseTime,
+                              systemClock.lastSyncTickCount,
+                              systemClock.slewRate,
+                              &systemClock.baseTime );
+
+        /* Calculate the new slew rate as offset in seconds of adjustment per second. */
+        systemClock.slewRate = clockOffsetSec / systemClock.pollPeriod;
+
+        systemClock.lastSyncTickCount = xTaskGetTickCount();
+    }
+
+    xSemaphoreGive( xMutex );
+}
+
+/*************************************************************************************/
+
+/**
+ * @brief Entry point of the demo that creates 2 tasks:
+ * 1. One task represents the SNTP client that periodically synchronizes system time with
+ * time from time servers.
+ * 2. The other task represents a sample application time that queries the system time
+ * periodically and prints it in human readable form of the YYYY-MM-DD hh:mm:ss.
+ */
+void vStartSntpDemo( void )
+{
+    /* On boot-up initialize the system time as the first second in the configured year. */
+    int32_t lNoOfYearsSince1900 = democonfigSYSTEM_START_YEAR - 1900;
+    int64_t llnoOfSecondsSince1900 = YEARS_TO_SECONDS( lNoOfYearsSince1900 );
+
+    if( llnoOfSecondsSince1900 <= UINT32_MAX )
+    {
+        systemClock.baseTime.secs = llnoOfSecondsSince1900;
+    }
+    else
+    {
+        systemClock.baseTime.secs = ( uint32_t ) llnoOfSecondsSince1900 - UINT32_MAX - 1UL;
+    }
+
+    LogInfo( ( "System time has been initialized to the year %u", democonfigSYSTEM_START_YEAR ) );
+    printTime( &systemClock.baseTime );
+
+    /* Initialize sempahore for guarding access to system clock variables. */
+    xMutex = xSemaphoreCreateMutexStatic( &xSemaphoreMutex );
+    configASSERT( xMutex );
+
+    /* Clear the first time sync completed flag of the system clock object so that a "step" correction
+     * of system time is utilized for the first time synchronization from a time server. */
+    systemClock.firstTimeSyncDone = false;
+
+    /* Create the SNTP client task that is reponsible for synchronizing system time with the time servers
+     * periodically. This is created as a high priority task to keep the SNTP client operation uninhindered. */
+    xTaskCreate( sntpTask,                 /* Function that implements the task. */
+                 "SntpClientTask",         /* Text name for the task - only used for debugging. */
+                 democonfigDEMO_STACKSIZE, /* Size of stack (in words, not bytes) to allocate for the task. */
+                 NULL,                     /* Task parameter - not used in this case. */
+                 configMAX_PRIORITIES - 1, /* Task priority, must be between 0 and configMAX_PRIORITIES - 1. */
+                 NULL );
+
+    /* Create the task that represents an application needing wall-clock time. */
+    xTaskCreate( prvSampleAppTask,         /* Function that implements the task. */
```

Comment:
Yeah, it can be helpful for large task application logic but the sample application task has a very small LoC footprint for the separate file to be significant.

Reply:
It was a fair amount of effort to re-factor the code into separate files, and I have pushed the changes.